Dev version of zb crashing on build

Reported by @Abdiramen

## Steps to reproduce

1. Run the first example in https://zb.256lights.llc/getting-started#first-steps using dev version (noted below)

## Expected behavior
Build succeeds

## Actual behavior

Backend panic:

```plain
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb7fcf3]

goroutine 1302 [running]:
sync.(*WaitGroup).Go.func1.1()
        /home/vimto/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.linux-amd64/src/sync/waitgroup.go:251 +0x45
panic({0xcd0460?, 0x14ffa20?})
        /home/vimto/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.linux-amd64/src/runtime/panic.go:860 +0x13a
unique.Handle[...].Value(...)
        /home/vimto/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.linux-amd64/src/unique/handle.go:27
zb.256lights.llc/pkg/internal/backend.equivalenceClass.toRealizationOutputReference(...)
        /home/vimto/zb/internal/backend/equivalence_class.go:44
zb.256lights.llc/pkg/internal/backend.(*builder).do-range4(...)
        /home/vimto/zb/internal/backend/realize.go:915
zb.256lights.llc/pkg/internal/backend.(*builder).do.Set[...].func10(...)
        /home/vimto/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.linux-amd64/src/maps/iter.go:28
zb.256lights.llc/pkg/internal/backend.(*builder).do(0x1aa5b17b31c0, {0xe1e050, 0x1aa5b19624b0}, {0x1aa5b195c700, 0x3c}, 0x1aa5b17a7d70, 0x0)
        /home/vimto/zb/internal/backend/realize.go:912 +0x28b3
zb.256lights.llc/pkg/internal/backend.(*builder).realize(0x1aa5b17b31c0, {0xe1e050, 0x1aa5b19624b0}, 0x1aa5b1837a40, 0x0)
        /home/vimto/zb/internal/backend/realize.go:384 +0x8e7
zb.256lights.llc/pkg/internal/backend.(*Server).realize.func1()
        /home/vimto/zb/internal/backend/realize.go:113 +0x3b2
sync.(*WaitGroup).Go.func1()
        /home/vimto/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.linux-amd64/src/sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 1300
        /home/vimto/go/pkg/mod/golang.org/toolchain@v0.0.1-go1.26.1.linux-amd64/src/sync/waitgroup.go:238 +0x73
```

## Version information
- zb version: 7428ce2666eca05cfe79e92781c79b6d028ea729
- Operating System and Version: `Linux 6.12.85+deb13-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.12.85-1 (2026-04-30)`
- CPU architecture:  x86_64-unknown-linux

I was able to reproduce this locally on c28692d54e916f74df556321373179b9bd4f3374. Relevant debug logs:

```plain
zb: 2026/05/17 17:12:51 DEBUG: Reached /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv
zb: 2026/05/17 17:12:51 DEBUG: Hashed /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv to sha256-E7vF8sqxzZakF3PIARooaSxc9nHFK03NE+tZ7JTtIvQ=
zb: 2026/05/17 17:12:51 DEBUG: Waiting for build lock on /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv...
zb: 2026/05/17 17:12:51 DEBUG: Acquired build lock on /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv
zb: 2026/05/17 17:12:51 DEBUG: Searching for realizations for sha256-E7vF8sqxzZakF3PIARooaSxc9nHFK03NE+tZ7JTtIvQ=!out...
zb: 2026/05/17 17:12:51 DEBUG: No suitable realizations exist for sha256-E7vF8sqxzZakF3PIARooaSxc9nHFK03NE+tZ7JTtIvQ=!out
zb: 2026/05/17 17:12:51 DEBUG: Waiting for lock on /opt/zb/store/6ssl5z26zmr9dn2iz4xi17a13ia7qz8y-gcc-4.2.1 (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)...
zb: 2026/05/17 17:12:51 DEBUG: /opt/zb/store/6ssl5z26zmr9dn2iz4xi17a13ia7qz8y-gcc-4.2.1 exists=true (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)
zb: 2026/05/17 17:12:51 DEBUG: Waiting for lock on /opt/zb/store/hpsxd175dzfmjrg27pvvin3nzv3yi61k-busybox-1.36.1 (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)...
zb: 2026/05/17 17:12:51 DEBUG: /opt/zb/store/hpsxd175dzfmjrg27pvvin3nzv3yi61k-busybox-1.36.1 exists=true (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)
zb: 2026/05/17 17:12:51 DEBUG: Waiting for lock on /opt/zb/store/yym06xrz2nq7na2gjkd907rn176b730a-wrapper.sh (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)...
zb: 2026/05/17 17:12:51 DEBUG: /opt/zb/store/yym06xrz2nq7na2gjkd907rn176b730a-wrapper.sh exists=true (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)
zb: 2026/05/17 17:12:51 DEBUG: Waiting for lock on /opt/zb/store/zl78sb229g3yp2ijayr21gcqn6n826x2-make-wrapper.sh (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)...
zb: 2026/05/17 17:12:51 DEBUG: /opt/zb/store/zl78sb229g3yp2ijayr21gcqn6n826x2-make-wrapper.sh exists=true (input to /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv)
zb: 2026/05/17 17:12:51 DEBUG: Using build user 30001:30000
zb: 2026/05/17 17:12:51 DEBUG: Runner for /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv is sandbox
zb: 2026/05/17 17:12:51 DEBUG: Output map for /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv: out -> /opt/zb/store/24882z36ywcycq71046cjkj8ysk8krj7-gcc-4.2.1
zb: 2026/05/17 17:12:51 DEBUG: Starting builder for /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv...
zb: 2026/05/17 17:12:51 DEBUG: Creating sandbox at /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot...
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /tmp/zb-build-gcc-4.2.1-879232903 /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/build
zb: 2026/05/17 17:12:51 DEBUG: mount -t tmpfs -o size=50% none /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/shm
zb: 2026/05/17 17:12:51 DEBUG: mount -t devpts -o newinstance,mode=0620 none /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/pts
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /dev/full /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/full
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /dev/null /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/null
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /dev/random /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/random
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /dev/tty /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/tty
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /dev/urandom /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/urandom
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /dev/zero /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/dev/zero
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /opt/zb/store/hpsxd175dzfmjrg27pvvin3nzv3yi61k-busybox-1.36.1 /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/opt/zb/store/hpsxd175dzfmjrg27pvvin3nzv3yi61k-busybox-1.36.1
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /opt/zb/store/yym06xrz2nq7na2gjkd907rn176b730a-wrapper.sh /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/opt/zb/store/yym06xrz2nq7na2gjkd907rn176b730a-wrapper.sh
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /opt/zb/store/zl78sb229g3yp2ijayr21gcqn6n826x2-make-wrapper.sh /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/opt/zb/store/zl78sb229g3yp2ijayr21gcqn6n826x2-make-wrapper.sh
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /opt/zb/store/6ssl5z26zmr9dn2iz4xi17a13ia7qz8y-gcc-4.2.1 /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/opt/zb/store/6ssl5z26zmr9dn2iz4xi17a13ia7qz8y-gcc-4.2.1
zb: 2026/05/17 17:12:51 DEBUG: ln -s /bin/dash /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/bin/sh
zb: 2026/05/17 17:12:51 DEBUG: mount --rbind /mnt/wsl/resolv.conf /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot/mnt/wsl/resolv.conf
zb: 2026/05/17 17:12:51 DEBUG: Created sandbox at /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv.chroot
zb: 2026/05/17 17:12:51 DEBUG: Builder for /opt/zb/store/szi8ijj8avl432ivwx8ncps8j8v1x52p-gcc-4.2.1.drv has finished successfully
zb: 2026/05/17 17:12:51 DEBUG: Processing floating output /opt/zb/store/24882z36ywcycq71046cjkj8ysk8krj7-gcc-4.2.1...
zb: 2026/05/17 17:12:51 DEBUG: Scanning for references in /opt/zb/store/24882z36ywcycq71046cjkj8ysk8krj7-gcc-4.2.1. Possible: {/opt/zb/store/6ssl5z26zmr9dn2iz4xi17a13ia7qz8y-gcc-4.2.1 /opt/zb/store/hpsxd175dzfmjrg27pvvin3nzv3yi61k-busybox-1.36.1 /opt/zb/store/yym06xrz2nq7na2gjkd907rn176b730a-wrapper.sh /opt/zb/store/zl78sb229g3yp2ijayr21gcqn6n826x2-make-wrapper.sh}
zb: 2026/05/17 17:12:51 DEBUG: Found references in /opt/zb/store/24882z36ywcycq71046cjkj8ysk8krj7-gcc-4.2.1 (self=false): {/opt/zb/store/6ssl5z26zmr9dn2iz4xi17a13ia7qz8y-gcc-4.2.1 /opt/zb/store/hpsxd175dzfmjrg27pvvin3nzv3yi61k-busybox-1.36.1}
zb: 2026/05/17 17:12:51 DEBUG: Determined /opt/zb/store/24882z36ywcycq71046cjkj8ysk8krj7-gcc-4.2.1 hashes to /opt/zb/store/3zm3n552583vwn238fp7d9i9f0cbgnr9-gcc-4.2.1, acquring lock...
zb: 2026/05/17 17:12:51 DEBUG: Acquired lock on /opt/zb/store/3zm3n552583vwn238fp7d9i9f0cbgnr9-gcc-4.2.1
zb: 2026/05/17 17:12:51 DEBUG: Moving /opt/zb/store/24882z36ywcycq71046cjkj8ysk8krj7-gcc-4.2.1 to /opt/zb/store/3zm3n552583vwn238fp7d9i9f0cbgnr9-gcc-4.2.1 (self-references=false)
zb: 2026/05/17 17:12:51 DEBUG: Registering metadata for /opt/zb/store/3zm3n552583vwn238fp7d9i9f0cbgnr9-gcc-4.2.1
zb: 2026/05/17 17:12:51 DEBUG: Marking /opt/zb/store/3zm3n552583vwn238fp7d9i9f0cbgnr9-gcc-4.2.1 read-only...
panic: runtime error: invalid memory address or nil pointer dereference [recovered, repanicked]
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xb5b153]

goroutine 1387 [running]:
sync.(*WaitGroup).Go.func1.1()
        /nix/store/bnv5q7irdqhp63620r8svjm2acfg9swl-go-1.26.1/share/go/src/sync/waitgroup.go:251 +0x45
panic({0xcda400?, 0x14f3d20?})
        /nix/store/bnv5q7irdqhp63620r8svjm2acfg9swl-go-1.26.1/share/go/src/runtime/panic.go:860 +0x13a
unique.Handle[...].Value(...)
        /nix/store/bnv5q7irdqhp63620r8svjm2acfg9swl-go-1.26.1/share/go/src/unique/handle.go:27
zb.256lights.llc/pkg/internal/backend.equivalenceClass.toRealizationOutputReference(...)
        /home/light/src/zb.256lights.llc/pkg/internal/backend/equivalence_class.go:44
zb.256lights.llc/pkg/internal/backend.(*builder).do-range4(...)
        /home/light/src/zb.256lights.llc/pkg/internal/backend/realize.go:915
zb.256lights.llc/pkg/internal/backend.(*builder).do.Set[...].func10(...)
        /nix/store/bnv5q7irdqhp63620r8svjm2acfg9swl-go-1.26.1/share/go/src/maps/iter.go:28
zb.256lights.llc/pkg/internal/backend.(*builder).do(0x25f99e68a000, {0xe1d670, 0x25f99e386af0}, {0x25f99e262680, 0x3c}, 0x25f99e2f0840, 0x0)
        /home/light/src/zb.256lights.llc/pkg/internal/backend/realize.go:912 +0x28b3
zb.256lights.llc/pkg/internal/backend.(*builder).realize(0x25f99e68a000, {0xe1d670, 0x25f99e386af0}, 0x25f99e0e8000, 0x0)
        /home/light/src/zb.256lights.llc/pkg/internal/backend/realize.go:384 +0x8e7
zb.256lights.llc/pkg/internal/backend.(*Server).realize.func1()
        /home/light/src/zb.256lights.llc/pkg/internal/backend/realize.go:113 +0x3b2
sync.(*WaitGroup).Go.func1()
        /nix/store/bnv5q7irdqhp63620r8svjm2acfg9swl-go-1.26.1/share/go/src/sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 1385
        /nix/store/bnv5q7irdqhp63620r8svjm2acfg9swl-go-1.26.1/share/go/src/sync/waitgroup.go:238 +0x73
```

--- a/internal/backend/equivalence_class.go
+++ b/internal/backend/equivalence_class.go
@@ -13,8 +13,8 @@ import (
 	"zombiezen.com/go/nix"
 )
 
-// equivalenceClass is an equivalence class of [zbstore.OutputReference] values.
-// It represents a single output of equivalent derivations.
+// equivalenceClass is a representation of [zbstore.RealizationOutputReference]
+// that implements [comparable].
 type equivalenceClass struct {
 	drvHashKey hashKey
 	outputName unique.Handle[string]
@@ -31,6 +31,9 @@ func newEquivalenceClass(drvHash nix.Hash, outputName string) equivalenceClass {
 }
 
 func realizationOutputReferenceKey(ref zbstore.RealizationOutputReference) equivalenceClass {
+	if ref.IsZero() {
+		return equivalenceClass{}
+	}
 	return newEquivalenceClass(ref.DerivationHash, ref.OutputName)
 }
 
@@ -39,6 +42,9 @@ func (eqClass equivalenceClass) isZero() bool {
 }
 
 func (eqClass equivalenceClass) toRealizationOutputReference() zbstore.RealizationOutputReference {
+	if eqClass.isZero() {
+		return zbstore.RealizationOutputReference{}
+	}
 	return zbstore.RealizationOutputReference{
 		DerivationHash: eqClass.drvHashKey.toHash(),
 		OutputName:     eqClass.outputName.Value(),

--- a/internal/backend/realize.go
+++ b/internal/backend/realize.go
@@ -665,6 +665,7 @@ func (b *builder) do(ctx context.Context, drvPath zbstore.Path, outputNames sets
 	if drvHash.IsZero() {
 		return fmt.Errorf("build %s: missing hash", drvPath)
 	}
+	drvHashKey := makeHashKey(drvHash)
 	for outputName := range outputNames.All() {
 		if drv.Outputs[outputName.Value()] == nil {
 			ref := zbstore.OutputReference{
@@ -698,7 +699,11 @@ func (b *builder) do(ctx context.Context, drvPath zbstore.Path, outputNames sets
 		// Search for existing realizations first.
 		wantEqClasses := sets.Collect(func(yield func(equivalenceClass) bool) {
 			for outputName := range outputNames.All() {
-				if !yield(newEquivalenceClass(drvHash, outputName.Value())) {
+				eqClass := equivalenceClass{
+					drvHashKey: drvHashKey,
+					outputName: outputName,
+				}
+				if !yield(eqClass) {
 					return
 				}
 			}
@@ -711,7 +716,10 @@ func (b *builder) do(ctx context.Context, drvPath zbstore.Path, outputNames sets
 		// Otherwise, we set a default of nulls for all requested outputs.
 		err = setBuildResultOutputs(conn, buildResultID, func(yield func(string, zbstore.Path) bool) {
 			for outputName := range outputNames.All() {
-				eqClass := newEquivalenceClass(drvHash, outputName.Value())
+				eqClass := equivalenceClass{
+					drvHashKey: drvHashKey,
+					outputName: outputName,
+				}
 				var path zbstore.Path
 				if reuseError == nil {
 					path = b.realizations[eqClass].path
@@ -898,7 +906,11 @@ func (b *builder) do(ctx context.Context, drvPath zbstore.Path, outputNames sets
 		}
 		delete(tempOutPaths, outputName) // No longer needs cleanup if we fail.
 
-		prev, previouslyRealized := b.realizations[newEquivalenceClass(drvHash, outputName)]
+		eqClass := equivalenceClass{
+			drvHashKey: drvHashKey,
+			outputName: unique.Make(outputName),
+		}
+		prev, previouslyRealized := b.realizations[eqClass]
 		if previouslyRealized && info.StorePath != prev.path {
 			// This should have been prevented at a higher level,
 			// but we do a safety check here anyway.

--- a/internal/backend/realize_test.go
+++ b/internal/backend/realize_test.go
@@ -563,6 +563,91 @@ func TestRealizeReferenceToDep(t *testing.T) {
 	}
 }
 
+func TestRealizeInputReference(t *testing.T) {
+	ctx := testcontext.New(t)
+	dir := backendtest.NewStoreDirectory(t)
+
+	const inputContent = "Hello, World!\n"
+	exportBuffer := new(bytes.Buffer)
+	exporter := zbstore.NewExportWriter(exportBuffer)
+	inputFilePath, _, err := storetest.ExportSourceFile(exporter, []byte(inputContent), storetest.SourceExportOptions{
+		Name:      "hello.txt",
+		Directory: dir,
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	const wantOutputName = "hello-path.txt"
+	drvContent := &zbstore.Derivation{
+		Name:   wantOutputName,
+		Dir:    dir,
+		System: system.Current().String(),
+		Env: map[string]string{
+			"in":  string(inputFilePath),
+			"out": zbstore.HashPlaceholder("out"),
+		},
+		InputSources: *sets.NewSorted(inputFilePath),
+		Outputs: map[string]*zbstore.DerivationOutputType{
+			zbstore.DefaultDerivationOutputName: zbstore.RecursiveFileFloatingCAOutput(nix.SHA256),
+		},
+	}
+	if runtime.GOOS == "windows" {
+		drvContent.Builder = powershellPath
+		drvContent.Args = []string{"-Command", "\"${env:in}`n\" | Out-File -NoNewline -Encoding ascii -FilePath ${env:out}"}
+	} else {
+		drvContent.Builder = shPath
+		drvContent.Args = []string{"-c", `echo "$in" > "$out"`}
+	}
+	drvPath, _, err := storetest.ExportDerivation(exporter, drvContent)
+	if err != nil {
+		t.Fatal(err)
+	}
+	if err := exporter.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	_, client, err := backendtest.NewServer(ctx, t, dir, &backendtest.Options{
+		TempDir: t.TempDir(),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	codec, releaseCodec, err := storeCodec(ctx, client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	err = codec.Export(nil, exportBuffer)
+	releaseCodec()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	realizeResponse := new(zbstorerpc.RealizeResponse)
+	err = jsonrpc.Do(ctx, client, zbstorerpc.RealizeMethod, realizeResponse, &zbstorerpc.RealizeRequest{
+		DrvPaths: []zbstore.Path{drvPath},
+	})
+	if err != nil {
+		t.Fatal("RPC error:", err)
+	}
+	if realizeResponse.BuildID == "" {
+		t.Fatal("no build ID returned")
+	}
+	got, err := backendtest.WaitForSuccessfulBuild(ctx, client, realizeResponse.BuildID)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	wantOutputContent := inputFilePath + "\n"
+	wantOutputPath, err := singleFileOutputPath(dir, wantOutputName, []byte(wantOutputContent), zbstore.References{
+		Others: *sets.NewSorted(inputFilePath),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+	checkSingleFileOutput(t, drvPath, wantOutputPath, []byte(wantOutputContent), got)
+}
+
 func TestRealizeSelfReference(t *testing.T) {
 	ctx := testcontext.New(t)
 	dir := backendtest.NewStoreDirectory(t)

--- a/zbstore/zbstore.go
+++ b/zbstore/zbstore.go
@@ -176,6 +176,11 @@ type RealizationOutputReference struct {
 	OutputName     string   `json:"outputName"`
 }
 
+// IsZero reports whether ref is the zero value.
+func (ref RealizationOutputReference) IsZero() bool {
+	return ref.DerivationHash.IsZero() && ref.OutputName == ""
+}
+
 // String returns the hash and the output name separated by "!".
 func (ref RealizationOutputReference) String() string {
 	return ref.DerivationHash.Base64() + "!" + ref.OutputName

